nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/util/SecureNiFiConfigUtil.java

```diff
@@ -0,0 +1,191 @@
+                            "This certificate will expire in {} days, and it is recommended to acquire your own certificates in order to properly secure Apache NiFi.", CERT_DURATION_DAYS);
+                    try {
+                        Pair<String[], String[]> subjectAlternativeNames = getSubjectAlternativeNames(nifiProperties);
+                        tlsConfiguration = KeyStoreUtils.createTlsConfigAndNewKeystoreTruststore(StandardTlsConfiguration
+                                .fromNiFiProperties(nifiProperties), CERT_DURATION_DAYS, subjectAlternativeNames.getLeft(),
+                                subjectAlternativeNames.getRight());
+                    } catch (IOException e) {
+                        cmdLogger.error("Encountered an I/O exception while generating secure Apache NiFi configuration.", e);
+                        throw e;
+                    } catch (GeneralSecurityException e) {
+                        cmdLogger.error("Encountered a security exception while generating secure Apache NiFi configuration.", e);
+                        throw new RuntimeException(e);
+                    }
+
+                    // Move over the new stores from temp dir
+                    Files.move(new File(tlsConfiguration.getKeystorePath()).toPath(), new File(keystorePath).toPath(),
+                            StandardCopyOption.REPLACE_EXISTING);
+                    Files.move(new File(tlsConfiguration.getTruststorePath()).toPath(), new File(truststorePath).toPath(),
+                            StandardCopyOption.REPLACE_EXISTING);
+
+                    updateProperties(propertiesFile, tlsConfiguration);
+
+                    cmdLogger.info("Successfully generated {} and {}.", keystorePath, truststorePath);
+                } else if (!keystore.exists() && truststore.exists()) {
+                    cmdLogger.error("Tried to generate keystore {} for secure Apache NiFi configuration, but truststore file {} already exists.  Aborting.",
+                            keystorePath, truststorePath);
+                    throw new RuntimeException("Will not generate keystore and truststore separately.");
+                } else if (keystore.exists() && !truststore.exists()) {
+                    cmdLogger.error("Tried to generate truststore {} for secure Apache NiFi configuration, but keystore file {} already exists.  Aborting.",
+                            truststorePath, keystorePath);
+                    throw new RuntimeException("Will not generate keystore and truststore separately.");
+                } else {
+                    cmdLogger.info("Existing keystore and truststore detected: skipping Apache Nifi certificate generation.");
+                }
+            } else {
+                cmdLogger.warn("HTTPS is configured, but keystore and truststore are not specified.  This will result in an invalid configuration.");
```

Comment:
This may be useful information, but it is probably better handled in other existing initialization methods.